tls/s2n_resume.c

```diff
@@ -768,3 +768,34 @@ int s2n_connection_add_new_tickets_to_send(struct s2n_connection *conn, uint8_t
 
     return S2N_SUCCESS;
 }
+
+int s2n_config_set_session_ticket_callback(struct s2n_config *config, s2n_session_ticket_cb cb)
+{
```

Comment:
We should add a check for cb not being null here: `POSIX_ENSURE_REF(cb);`

Reply:
I thought we only needed to do that for pointers?

Reply:
We need a way to be able to unset callbacks so we can't ensure ref

Reply:
Your typedef hides it here, but technically cb is a pointer. A pointer to a function :)

Reply:
s2n_session_ticket_cb is a pointer and this API expects the cb to be notnull here.